### PR TITLE
PEtab-multilanguage proof of concept for PySB-PEtab

### PR DESCRIPTION
**Pull request by dweindl**
This implements a proof of concept for the PEtab [multi-language proposal](https://github.com/PEtab-dev/PEtab/pull/538). 
Handles PySB-PEtab problems using the recent PEtab model abstraction (https://github.com/PEtab-dev/libpetab-python/pull/145). 
Removes the (need for the) current `PysbPetabProblem` and allows for importing PEtab problems through a single interface.

Runs the tentative PEtab version 2 test cases from https://github.com/PEtab-dev/petab_test_suite/pull/53.

**Blocked by https://github.com/pysb/pysb/pull/570**
~~**Blocked by https://github.com/PEtab-dev/libpetab-python/pull/203**~~

### CODE REVIEW

File: .github/workflows/test_petab_test_suite.yml [hunk 1]
```diff
@@ -70,6 +73,7 @@ jobs:
       - name: Run PEtab-related unit tests
         run: |
           source ./build/venv/bin/activate \
+           && pip install --force-reinstall -v git+https://github.com/PEtab-dev/libpetab-python.git@develop \
```
Comment (dweindl) [inline .github/workflows/test_petab_test_suite.yml:76]:
```suggestion
```

Before merge

File: scripts/installAmiciSource.sh [hunk 2]
```diff
@@ -30,6 +30,7 @@ fi
 pip install -U "setuptools<64"
 pip install --upgrade pip wheel
 pip install --upgrade pip scipy matplotlib coverage pytest \
-   pytest-cov cmake_build_extension numpy
-pip install --verbose -e ${AMICI_PATH}/python/sdist[petab,test,pysb,vis] --no-build-isolation
+  pytest-cov cmake_build_extension numpy
+pip install git+https://github.com/FFroehlich/pysb@fix_pattern_matching # pin to PR for SPM with compartments
+pip install --verbose -e ${AMICI_PATH}/python/sdist[petab,test,vis] --no-build-isolation
```
Comment (dweindl) [inline scripts/installAmiciSource.sh:35]:
```suggestion
   pytest-cov cmake_build_extension numpy
pip install --verbose -e ${AMICI_PATH}/python/sdist[petab,test,pysb,vis] --no-build-isolation
```
before merge

File: documentation/rtd_requirements.txt [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # NOTE: relative paths are expected to be relative to the repository root
+#git+https://github.com/PEtab-dev/libpetab-python@develop#egg=petab
```
Comment (FFroehlich) [inline documentation/rtd_requirements.txt:2]:
```suggestion
petab>=0.2.1
```

**Review by FFroehlich — APPROVED** [review]
:+1:

File: documentation/rtd_requirements.txt (hunk 3 above)
Comment (dweindl) [reply documentation/rtd_requirements.txt:2]: The current requirement is handled via setup.cfg. Will remove.

**Pull request merged**